src/redistributors/morpho/Distributor.sol

```diff
@@ -4,91 +4,135 @@ pragma solidity 0.8.23;
 
 import { ERC20Helper } from "../../../lib/erc20-helper/src/ERC20Helper.sol";
 
-import { IMorphoVaultFactoryLike, IMorphoVaultLike, IWrappedMLike } from "./Dependencies.sol";
+import {
+    Id as MarketId,
+    IMorphoBlueLIke,
+    IMorphoVaultFactoryLike,
+    IMorphoVaultLike,
+    IWrappedMLike
+} from "./Dependencies.sol";
+
+import { IDistributor } from "./interfaces/IDistributor.sol";
 
 import { AccountFundsDistribution } from "./AccountFundsDistribution.sol";
 import { VaultFundsDistribution } from "./VaultFundsDistribution.sol";
 
+// TODO: Permit-based deposit.
+
 /**
  * @title Morpho Distributor (Dual-Cascade, Cumulative Distribution)
  * @dev   A contract for distributing Morpho yield across accounts within vaults.
  * @dev   Each distribution cascade is an ERC2222 model relying on tracking per-account "corrections".
  */
-contract Distributor is AccountFundsDistribution, VaultFundsDistribution {
-    error InvalidVault();
-    error TransferFailed();
-    error TransferFromFailed();
+contract Distributor is IDistributor, AccountFundsDistribution, VaultFundsDistribution {
+    /* ============ Variables ============ */
 
+    uint256 internal constant _MORPHO_POSITION_SLOT = 2;
+    uint256 internal constant _MORPHO_MARKET_SLOT = 3;
+
+    /// @inheritdoc IDistributor
     address public immutable wrappedMToken;
+
+    /// @inheritdoc IDistributor
     address public immutable morphoBlue;
+
+    /// @inheritdoc IDistributor
     address public immutable morphoVaultFactory;
 
     uint256 internal _lastWrappedMBalance;
 
     mapping(address vault => uint256 lastCumulativeDistribution) internal _lastCumulativeDistributions;
     mapping(address vault => mapping(address account => uint256 claimed)) internal _claims;
 
+    /* ============ Constructor ============ */
+
     constructor(address wrappedMToken_, address morphoBlue_, address morphoVaultFactory_) {
         wrappedMToken = wrappedMToken_;
         morphoBlue = morphoBlue_;
         morphoVaultFactory = morphoVaultFactory_;
     }
 
-    /// @notice Allow `msg.sender` to deposit an amount of Morpho Vault shares to get access to WrappedM yield.
-    function deposit(address vault_, uint256 vaultShares_) external {
+    /* ============ Interactive Functions ============ */
+
+    /// @inheritdoc IDistributor
+    function deposit(address vault_, uint256 amount_) external {
         _revertIfNotMorphoVault(vault_);
 
-        if (!ERC20Helper.transferFrom(vault_, msg.sender, address(this), vaultShares_)) revert TransferFromFailed();
+        emit Deposited(vault_, msg.sender, amount_);
+
+        if (!ERC20Helper.transferFrom(vault_, msg.sender, address(this), amount_)) revert TransferFromFailed();
```

Comment:
Depends what's more important: CEI or the Deposited/Transfer event order. Since the vault is a 3rd party contract, I would rather not assume their transfer fucntion does not do other stuff, and just follow CEI strictly.